src/components/CalendarPicker/calendarPickerPropTypes.js

```diff
@@ -0,0 +1,23 @@
+import PropTypes from 'prop-types';
+
+const propTypes = {
+    /** A function that is called when the day is clicked */
+    onChange: PropTypes.func.isRequired,
+
+    /** A value initial of date */
+    value: PropTypes.objectOf(Date),
+
+    /** A minimum date of calendar to select */
+    minDate: PropTypes.objectOf(Date),
+
+    /** A maximum date of calendar to select */
+    maxDate: PropTypes.objectOf(Date),
```

Comment:
I suggest sticking with the standard JS Date type, as it's used for most date pickers libraries, instead of complicating the complexity of the parameters and handling more edge cases.